src/app/clusters/time-synchronization-server/time-synchronization-delegate.h

```diff
@@ -104,12 +104,17 @@ class Delegate
      * a CHIP_ERROR.
      */
     virtual CHIP_ERROR UpdateTimeUsingNTPFallback(const CharSpan & fallbackNTP,
-                                                  chip::Callback::Callback<OnFallbackNTPCompletion> * callback) = 0;
+                                                  chip::Callback::Callback<OnFallbackNTPCompletion> * callback)
+    {
+        return CHIP_ERROR_NOT_IMPLEMENTED;
+    }
 
     /**
      * @brief Signals application that UTCTime has changed through the timesync cluster.
      */
-    virtual void UTCTimeAvailabilityChanged(uint64_t time) = 0;
+    virtual void UTCTimeAvailabilityChanged(uint64_t time) {}
```

Comment:
Is there a way for time to go from available -> unavailable? ie, does the time need to be optional or something?

Can you add in the description when this is expected to be called? ie, are you guaranteed to get a call to this function even if the time comes from an internal source?

Reply:
The name of the function might be the confusing aspect here since this gets called only when time is set through SetUTCTime(). I will add a description.

Reply:
Also, I don't see a way to get rid of time once it is set. That is why I didn't consider making this optional. When there is a time failure, there is another function to notify this to the app - NotifyTimeFailure().